.github/workflows/spectral-validation.yml

```diff
@@ -0,0 +1,23 @@
+name: Run Spectral Validation on all samples
+
+on:
+  - pull_request
```

Comment:
```suggestion
  - pull_request
   - push:
      branches:
        - main
```

Reply:
Should we also not run this again main when merged? (perhaps over kill but also nice to know the main branch is correct:) )

Reply:
Yeah I think so.
We should run this 
 - when a PR is raised
 - when any new commits are pushed to said PR
 - when a PR is merged